sdk/javascript/src/symbol/VotingKeysGenerator.js

```diff
@@ -0,0 +1,65 @@
+const { KeyPair } = require('./KeyPair');
+const { PrivateKey } = require('../CryptoTypes');
+
+const setBuffer = (destination, offset, source) => {
+	source.forEach((byte, i) => { destination.setUint8(offset + i, source[i]); });
+};
+
+/**
+ * Generates symbol voting keys.
+ */
+class VotingKeysGenerator {
+	/**
+	 * Creates a generator around a voting root key pair.
+	 * @param {KeyPair} rootKeyPair Voting root key pair.
+	 * @param {function} privateKeyGenerator Private key generator.
+	 */
+	constructor(rootKeyPair, privateKeyGenerator = PrivateKey.random) {
+		this.rootKeyPair = rootKeyPair;
+		this.privateKeyGenerator = privateKeyGenerator;
+	}
+
+	/**
+	 * Generates voting keys for specified epochs.
+	 * @param {number} startEpoch Start epoch.
+	 * @param {number} endEpoch End epoch.
+	 * @returns {Uint8Array} Serialized voting keys.
+	 */
+	generate(startEpoch, endEpoch) {
+		const HEADER_SIZE = 80;
+		const EPOCH_ENTRY_SIZE = 96;
+
+		const numEpochs = Number(endEpoch - startEpoch) + 1;
+		const buffer = new ArrayBuffer(HEADER_SIZE + (EPOCH_ENTRY_SIZE * numEpochs));
+
+		const view = new DataView(buffer);
+		view.setBigUint64(0, BigInt(startEpoch), true);
+		view.setBigUint64(8, BigInt(endEpoch), true);
+		view.setBigUint64(16, 0xFFFFFFFFFFFFFFFFn, true);
+		view.setBigUint64(24, 0xFFFFFFFFFFFFFFFFn, true);
+
+		setBuffer(view, 32, this.rootKeyPair.publicKey.bytes);
+		view.setBigUint64(64, BigInt(startEpoch), true);
+		view.setBigUint64(72, BigInt(endEpoch), true);
```

Comment:
hmm, maybe we should ad comment here, what this fields are
(or some simple in-code explanation/ascii-drawing of how the format looks like)